### PR TITLE
feat: view chart query

### PR DESCRIPTION
**Pull request by rephus**
Closes: #4591

### Description:

Creating new endpoint for `/runViewChartQuery` 

I had to refactor the existing ExploreProvider to take a different hook for making the request, depending on the edit mode. 

Screenshot from 2023-03-22 10-24-44
Screenshot from 2023-03-22 10-25-12

### CODE REVIEW

File: packages/backend/src/routers/projectRouter.ts [hunk 1]
```diff
@@ -267,6 +267,41 @@ projectRouter.post(
...
+projectRouter.post(
+    '/explores/:exploreId/runViewChartQuery',
+    allowApiKeyAuthentication,
+    isAuthenticated,
+    async (req, res, next) => {
+        try {
+            const { body } = req;
+            const { csvLimit } = body;
+            const metricQuery: MetricQuery = {
+                dimensions: body.dimensions,
```
Comment (rephus) [inline packages/backend/src/routers/projectRouter.ts:279]: Same arguments as /runQuery

File: packages/frontend/src/pages/Explorer.tsx [hunk 2]
```diff
@@ -47,12 +48,21 @@ const ExplorerPage = memo(() => {
...
+
+    const queryResults = useQueryResults();
+
     if (user.data?.ability?.cannot('view', 'Project')) {
         return <ForbiddenPanel />;
     }
+
     return (
-        <ExplorerProvider isEditMode={true} initialState={explorerUrlState}>
+        <ExplorerProvider
```
Comment (rephus) [inline packages/frontend/src/pages/Explorer.tsx:61]: Splitting this component, will break the `save` transition, and trigger extra requests , like In here . So not worth it for now.
[Screencast from 21-03-23 14:53:35.webm](https://user-images.githubusercontent.com/1983672/226861449-7031367e-311b-433d-be36-5815e290af30.webm)

File: packages/frontend/src/pages/SavedExplorer.tsx [hunk 3]
```diff
@@ -53,6 +55,7 @@ const SavedExplorer = () => {
 
     return (
         <ExplorerProvider
```
Comment (rephus) [inline packages/frontend/src/pages/SavedExplorer.tsx:57]: Splitting this component, will break the save transition, and trigger extra requests , like In here . So not worth it for now.
[Screencast from 21-03-23 14:53:35.webm](https://user-images.githubusercontent.com/1983672/226861449-7031367e-311b-433d-be36-5815e290af30.webm)

**Review by owlas — COMMENTED** [review]
Is this going to work if the new endpoint accepts different parameters as the original `runQuery` endpoint?

For example `/runQuery` will accept a `MetricQuery` (since interactive viewers are freely allowed to change those), whereas `/runViewOnlyQuery` will probably just accept the chart id

**Comment by rephus** [comment]
> Probably, but that will require a bit of extra refactoring, so to keep things small and simple, I have just copied the method 

But I made a new ticket to refactor this on a separate PR https://github.com/lightdash/lightdash/issues/4892

**Comment by owlas** [comment]
got it!

**Pull request merged**